src/solver/variable/include/antares/solver/variable/economy/dispatchableGeneration.h

```diff
@@ -169,84 +125,95 @@ class DispatchableGeneration
     };
 
 public:
+    DispatchableGeneration():
+        pValuesForTheCurrentYear(nullptr)
+    {
+    }
+
     ~DispatchableGeneration()
     {
+        for (unsigned int numSpace = 0; numSpace < pNbYearsParallel; numSpace++)
+        {
+            delete[] pValuesForTheCurrentYear[numSpace];
+        }
         delete[] pValuesForTheCurrentYear;
     }
 
-    void initializeFromStudy(Data::Study& study)
+    void initializeFromArea(Data::Study* study, Data::Area* area)
     {
-        pNbYearsParallel = study.maxNbYearsInParallel;
+        pNbYearsParallel = study->maxNbYearsInParallel;
+        pValuesForTheCurrentYear = new VCardType::IntermediateValuesBaseType[pNbYearsParallel];
 
-        InitializeResultsFromStudy(AncestorType::pResults, study);
+        // Building the vector of group names the clusters belong to.
+        std::set<std::string> names;
```

Comment:
`std::set<std::string> names;` : it dos not seem we need it.
We could directly fill **groupNames_**.

Reply:
We need unique sorted names, `std::set` is a way to achieve this